packages/server/src/core/internals/procedureBuilder.ts

```diff
@@ -265,10 +265,16 @@ function isPlainObject(obj: unknown) {
   return obj && typeof obj === 'object' && !Array.isArray(obj);
 }
 
+/**
+ * @internal
+ * Please note, `trpc-openapi` uses this function.
+ */
 export function createInputMiddleware<T>(
   parse: ParseFn<T>,
 ): ProcedureBuilderMiddleware {
   return async function inputMiddleware({ next, rawInput, input }) {
+    (inputMiddleware as any)._input = true;
```

Comment:
```suggestion
    (inputMiddleware as any)._type = 'input';
```

And we should add `ProcedureBuilderMiddleware.type:string | undefined`

Reply:
LGTM 👍